internal/lm/fabric.go

```diff
@@ -0,0 +1,156 @@
+/**
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package lm
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"math/rand" // nolint:gosec
+	"net"
+	"os"
+	"sort"
+	"strings"
+
+	spec "github.com/NVIDIA/k8s-device-plugin/api/config/v1"
+	"github.com/NVIDIA/k8s-device-plugin/internal/resource"
+
+	"github.com/google/uuid"
+	"k8s.io/klog/v2"
+)
+
+func newImexDomainLabeler(config *spec.Config, devices []resource.Device) (Labeler, error) {
+	if config.Flags.GFD.ImexNodesConfigFile == nil {
+		// No imex config file, return empty labels
+		return empty{}, nil
+	}
+
+	imexConfigFile, err := os.Open(*config.Flags.GFD.ImexNodesConfigFile)
```

Comment:
So this file is assumed to be an absolute path, and not relative to the injected `/etc/nvidia-imex/` path? I don't feel too strongly either way, I just want to make sure this is a conscious decision.

Reply:
Also -- where is the default value of this being set? It used to be in the command line arguments, but I don't see that change in this diff anymore.

Reply:
For now, and until we face feedback that proves this wrong, I think, having `/etc/nvidia-imex` as the path inside the container serve us well. we then simply take a diff name than default via the configMap options `imexNodesConfigFile` Key.

Reply:
I don't think the path `/etc/nvidia-imex` inside the container should ever change. My question was about the config option begin reuired to be an absolute path of `/etc/nvidia-imex/nodes_config.cfg` vs. just `nodes_config.cfg` where we *always* prepend `/etc/nvidia-imex/` to it.

Reply:
actually, since we are defining a fixed path inside the container, how about I add this 

```go
	if config.Flags.GFD.ImexNodesConfigFile == nil {
		// No imex config file, return empty labels
		return empty{}, nil
	}

	file := path.Join("/etc/nvidia-imex", *config.Flags.GFD.ImexNodesConfigFile)
	imexConfigFile, err := os.Open(file)
```

This way we will only create the labels if an entry like in the Helm values
```yaml
config:
  map:
    default: |-
      version: v1
      flags:
        gfd:
          imexNodesConfigFile: file-name.cgf
```

Reply:
This is exactly what I am proposing.

Reply:
It still doesn't answer my second question about the default value. Where are we setting the default of `nodes_config.cfg`?

Reply:
I was thinking on that, we could add `nodes_config.cfg` as a hard coded default at https://github.com/NVIDIA/k8s-device-plugin/blob/main/api/config/v1/config.go#L111 

```go

func parseConfigFrom(reader io.Reader) (*Config, error) {
	var err error
	var configYaml []byte

	configYaml, err = io.ReadAll(reader)
	if err != nil {
		return nil, fmt.Errorf("read error: %v", err)
	}

	var config Config
	err = yaml.Unmarshal(configYaml, &config)
	if err != nil {
		return nil, fmt.Errorf("unmarshal error: %v", err)
	}

	if config.Version == "" {
		config.Version = Version
	}

	if config.Version != Version {
		return nil, fmt.Errorf("unknown version: %v", config.Version)
	}

	if config.Flags.GFD.ImexNodesConfigFile == nil {
		*config.Flags.GFD.ImexNodesConfigFile = "nodes_config.cfg"
	}

	return &config, nil
}
```

Reply:
As I mentioned above -- we should probably just make it the default value of the flag, (which essentially fills it in the config when it is parsed). We used to have that in this PR, but it seems dropped in the most recent update.

Reply:
based on https://github.com/NVIDIA/k8s-device-plugin/pull/965#discussion_r1782610125 it makes no sense to have CLI flag, just the entry on the ConfigMap Struct.

Reply:
This is what I'm referring to:
https://github.com/NVIDIA/k8s-device-plugin/pull/965#discussion_r1782628112

Reply:
Ok, so we add a flag and default value, but we don't create a way to mod said flag via Helm values, only via the ConfigMap